src/curve.ts

```diff
@@ -62,13 +79,23 @@ export class Curve extends Element {
   /**
    * @return {boolean} Returns true if this is a partial bar.
    */
-  isPartial() {
+  isPartial(): boolean {
     return !this.from || !this.to;
   }
 
-  renderCurve(params) {
-    const ctx = this.context;
+  renderCurve(params: CurveRenderParams): void {
+    const ctx = this.checkContext();
     const cps = this.render_options.cps;
+    if (
```

Comment:
I'd move all these checks into an internal `check()` method and call it form `renderCurve` and from `draw`.

```typescript
check() {
  if cps == undefined || cps[0] == undefined || ... throw new Vex.RERR("InvalidParams", "No curve control points");
  if ths.render_options.x_shift == undefined || y_shift == undefied ... throw new Vex.RERR("...
  ...
}
```

Reply:
If we do not make the check inside the method using the variable and we do it in another method that we call the compiler does not realise it and we still get an error. I believe that the checks will no longer be so complex defining `CurveOptions` as suggested below.

Reply:
As expected check no longer required